notify/notify.go

```diff
@@ -889,11 +891,15 @@ func (tms TimeMuteStage) Exec(ctx context.Context, l log.Logger, alerts ...*type
 		return ctx, alerts, errors.New("missing now timestamp")
 	}
 
-	muted, err := inTimeIntervals(now, tms.Times, muteTimeIntervalNames)
+	muted, mutedBy, err := inTimeIntervals(now, tms.Times, muteTimeIntervalNames)
 	if err != nil {
 		return ctx, alerts, err
 	}
-
+	for _, a := range alerts {
+		// If the alert is not muted then mutedBy is a nil slice. This will set the
+		// alerts back to active.
+		tms.marker.SetMuted(a.Fingerprint(), mutedBy...)
+	}
```

Comment:
with #3556 now merged we can remove all of the changes done to `notify.go` - we'll have to change the interface slightly because it now needs to accept `a.Fingerprint()` but that's ok given it is only used here.

Reply:
I'm not sure if `a.Fingerprint()` is sufficient as time mutes are applied against the route, not the alert. We can pass a list of fingerprints, or a list of alerts. I started doing this:

```go
type TimeMuter interface {
	Mutes(timeIntervalName []string, now time.Time, alerts ...Alert) (bool, error)
}
```

Reply:
Yeah, either can work! I chose a single one, because at least it's consistent with the other similar interface https://github.com/prometheus/alertmanager/blob/acb58400fd2b324dd4cc56110c8e891e8b1c0420/types/types.go#L380-L382

However, I just realised that what we want is `a.Labels` as opposed to `a.Fingerprint()` so that would be my suggestion.